src/main/java/io/netty/incubator/codec/quic/QuicChannelConfig.java

```diff
@@ -72,4 +72,17 @@
      * connection is established.</strong>
      */
     QuicChannelConfig setPeerCertServerName(String peerCertServerName);
+
+    /**
+     * Returns file path where cryptographic secrets are logged to.
+     */
+    String getKeylogPath();
+
+    /**
+     * Enables cryptographic secrets to be logged to the specified file path.
+     *
+     * <strong>Be aware this config setting can only be adjusted before the
+     * connection is established.</strong>
+     */
+    QuicChannelConfig setKeylogPath(String keylogPath);
```

Comment:
I wonder if we should only support this via `ChannelOption` for now as this seems to be very specific to quiche.

Reply:
I wouldn't say it's Quiche specific. It's an OpenSSL/BoringSSL API (e.g. [here](https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_keylog_callback.html)) and it's  commonly used it various tools. What I've been thinking about, Quiche internally handles TLS configuration separately from an application config. Maybe we need to do the same and separate out TLS-related settings like it's done with `SSLContext`? Including certs, ALPN, keyloggers etc

Reply:
I an ideal world we would be able to just submit the SSL pointer somehow to quiche and be able to reuse what we have in netty

Reply:
Seems like Quiche uses another Rust package to deal with SSL context, let me check the code to see if this is doable